Source/ID3TagEditor.swift

```diff
@@ -68,10 +70,9 @@ public class ID3TagEditor {
      ID3 tag).
      */
     public func write(tag: ID3Tag, to path: String, andSaveTo newPath: String? = nil) throws {
-        let mp3 = try mp3FileReader.readFileFrom(path: path)
-        let currentTag = try self.id3TagParser.parse(mp3: mp3)
-        let mp3WithId3Tag = try mp3WithID3TagBuilder.build(mp3: mp3, newId3Tag: tag, currentId3Tag: currentTag)
-        try mp3FileWriter.write(mp3: mp3WithId3Tag, path: newPath ?? path)
+        let currentId3Tag = try read(from: path)
```

Comment:
It seems that the problem rely in this change. If you rollback to 

```
        let mp3 = try mp3FileReader.readFileFrom(path: path)
        let currentId3Tag = try self.id3TagParser.parse(mp3: mp3)
```

Everything should be fine. I will investigate deeply after the merge if there are problems with the examples (because generally speaking other library are not so strict in the check as this one I did, and some of the mp3 where generated using them. So probably, when I created this example, they came as output from another library).

Can you verify that by reverting this part your implementation is still fine? This will be a BIG improvements for the library (thanks again for the contribution 🙏 )

Reply:
`readFileFrom(path:)` reads the whole file into memory, so I can't just revert that line. However, I think the problem is that `read(from:)` doesn't check for the ID3 tag presence first using `isTagPresentIn(mp3:version:). So it tries to parse the tag size even though there's no ID3 tag at all. I'll fix it.